### DISCUSSION TITLE
要是有域名自动解析就好了

### DISCUSSION THREAD

**Discussion by wwwliutao125** [XIU2/CloudflareSpeedTest#355] [讨论]
在路由器上跑，把最快的几个ip解析到阿里云，可惜自己太菜不会搞

**Comment by XIU2** [comment]
这你需要去找有没有 阿里云 的**自动更新域名 DNS 解析**的脚本或工具。
阿里云应该提供的有 API，Cloudflare 就提供的有 API 可以通过 POST 来添加/更新/删除域名解析记录。

我个人不使用阿里云，所以并不清楚。

****

我写软件很有洁癖或者强迫症，不会一股脑把什么都加进去，所以这个软件**不会去添加这类功能**，只会专注于测速 Cloudflare CDN 获得最快 IP 这一点。

对于这类功能，应该由其他项目、工具来解决。

**Comment by wwwliutao125** [comment]
v2ray套cdn翻墙的时候，这个ip也是优选的，过一段时间就得换一下。麻烦。要是脚本里集成这个功能，就永远不用动了，完美

**Comment by wwwliutao125** [comment]
我自己菜鸟不会弄，你考虑搞搞，大神

**Reply by Mrered** [reply]
看看这个[jeessy2/ddns-go: 简单好用的DDNS。自动更新域名解析到公网IP(支持阿里云、腾讯云dnspod、Cloudflare、Callback、华为云、百度云、Porkbun、GoDaddy、Google Domain) (github.com)](https://github.com/jeessy2/ddns-go)

**Comment by XIU2** [comment]
域名商太多，我不可能全都去研究一遍。

**Comment by wwwliutao125** [comment]
开发一个openwrt插件，对，最完美

**Comment by XIU2** [comment]
我不是万能的啊，而且我也没有 openwrt 相关路由器设备。

**Comment by wwwliutao125** [comment]
openwrt里面有个动态ddns，两个一结合最完美

**Comment by XIU2** [comment]
DDNS 是定期将当前宽带的公网 IP 解析到指定域名下。
这都是内置的，本质也是用 API 去更新域名解析记录。

**Comment by wwwliutao125** [comment]
说错了，dns

**Comment by wwwliutao125** [comment]
![image](https://user-images.githubusercontent.com/20475019/105627823-0c450800-5e74-11eb-8b9d-59ebccf1c6aa.png)

**Comment by wwwliutao125** [comment]
![image](https://user-images.githubusercontent.com/20475019/105627844-25e64f80-5e74-11eb-9e4b-2bb87aeb6221.png)

**Comment by XIU2** [comment]
最重要的是，我如果要去写一个 阿里云 自动更新域名解析记录的脚本或工具。
那么我首先要在 阿里云 买个域名，然后去阿里云查文档并测试，才能写出来。

其他域名注册商同理。

**Comment by wwwliutao125** [comment]
不用，你只要
![image](https://user-images.githubusercontent.com/20475019/105627859-40b8c400-5e74-11eb-867e-f77006dbbcc9.png)

**Comment by wwwliutao125** [comment]
只要有一个html文件，ip出来，就能和这个结合，结果别保存xls文档

**Comment by wwwliutao125** [comment]
我是瞎说，只供参考，haha

**Comment by wwwliutao125** [comment]
![image](https://user-images.githubusercontent.com/20475019/105627984-0ac80f80-5e75-11eb-9258-475197a504d5.png)

**Comment by XIU2** [comment]
可以写个简单的脚本来运行完软件后从结果文件中获取最快的 IP。

一种是来源 [脚本] 直接脚本输出最快的 IP。
一种是来源 [URL] 脚本输出最快的 IP 到文件，你再去整成 URL（这需要路由器上有 WEB 服务）。

你要哪个，另外我没用过你这种功能，所以我写脚本无法自己测试效果，可能还需要你来测试。

**Comment by wwwliutao125** [comment]
一种是来源 [URL] 脚本输出最快的 IP 到文件，你再去整成 URL（这需要路由器上有 WEB 服务）。这个通用，openwrt都有web服务

**Comment by XIU2** [comment]
简单的来说就是，脚本去运行我这个软件，然后再去从结果文件中获得最快的 IP，输出到一个 txt 文件。

这个 txt 文件要放在 WEB 服务目录下，这样你就能通过 URL 来访问这个文件了。

**Comment by wwwliutao125** [comment]
那几个ip要一个地址一个如http://www.xxx.com/1,http://www.xxx.com/2,http://www.xxx.com/3,，只要能这样出来，就能和刚刚动态dns结合

**Comment by wwwliutao125** [comment]
你是大神，我也不会，我连怎么你这个在openwrt下运行都不会。只会在windows上跑，然后再去改

**Comment by wwwliutao125** [comment]
路由器上自动的最好，图形界面也没啥必要

**Comment by XIU2** [comment]
这些都没问题，但最大的问题就是你可能不会用，搞得我都不知道该如何跟你说。

**Comment by wwwliutao125** [comment]
我能听懂你说的，我只是go语言不会编译，要不就直接改你源码了，对路由器也不是很熟悉，你要搞一个通用的肯定能火，需要的人很多

**Comment by XIU2** [comment]
比如你首先要去试试，你的路由器上面要运行哪个版本的 CloudflareST。

应该就是这两个之一。
CloudflareST_linux_arm.zip
CloudflareST_linux_arm64.zip

下载并解压后，放到你的路由器上面，然后运行：
```
chmod +x CloudflareST
./CloudflareST
```

****

至于通用，则不可能的，不同需求，不同系统，不同平台。

**Comment by wwwliutao125** [comment]
主机型号 | Phicomm N1
-- | --
架构 | ARMv8 Processor rev 4 (v8l) x 4
固件版本 | OpenWrt R20.7.1 / LuCI Master (git-20.117.60969-420c61a)

**Comment by XIU2** [comment]
不需要给我看这个，路由器基本上都是 ARM 架构，你要去试试用的是 32 位的还是 64 位的 CloudflareST。

**Comment by wwwliutao125** [comment]
github,有人有相同需求，https://github.com/badafans/better-cloudflare-ip

**Comment by wwwliutao125** [comment]
![image](https://user-images.githubusercontent.com/20475019/105628235-cb022780-5e76-11eb-976e-4aaacbf57bfc.png)

**Comment by wwwliutao125** [comment]
> 不需要给我看这个，路由器基本上都是ARM架构，你要去试试用的是32位的还是64位的CloudflareST。

我不会看啊，我又不是大神

**Comment by wwwliutao125** [comment]
哎

**Comment by wwwliutao125** [comment]
n1盒子，你搜搜这个

**Comment by XIU2** [comment]
我这个支持路由器里运行。

你这样的话，我总不能意念施法吧。。。

**Comment by XIU2** [comment]
你会玩 Linux 吗？
你这种路由器本质就是个微型 Linux 系统。

**Comment by wwwliutao125** [comment]
会一丁点点

**Comment by wwwliutao125** [comment]
很少一点点

**Comment by wwwliutao125** [comment]
等会，我看看是多少位的

**Comment by wwwliutao125** [comment]
![image](https://user-images.githubusercontent.com/20475019/105628338-45cb4280-5e77-11eb-9357-2754c197be66.png)

**Comment by wwwliutao125** [comment]
ARMv8 Processor rev 4 (v8l) x 4
--


是不是和这个有关

**Comment by XIU2** [comment]
这种功能如果是我自己的需求的话，我几分钟就搞完了。
但是跟你说的话，就还要先教会你怎么玩才行。

既然你说你会 Linux，那么：
你把 CloudflareST 放到到路由器上面。
然后赋予执行权限，再去运行。

```
chmod +x CloudflareST
./CloudflareST
```

能运行就说明能用。

**Comment by wwwliutao125** [comment]
好，你别急，教会我，就能教会很多人

**Comment by wwwliutao125** [comment]
等

**Comment by wwwliutao125** [comment]
![image](https://user-images.githubusercontent.com/20475019/105628417-bc684000-5e77-11eb-972a-7da71acadda5.png)

**Comment by wwwliutao125** [comment]
怎么放呢，哎

**Comment by XIU2** [comment]
干脆我远程协助吧。

**Comment by wwwliutao125** [comment]
成啊，嘿嘿，谢谢

**Comment by wwwliutao125** [comment]
咋远程呢，你可以搞个教程，肯定火

**Comment by wwwliutao125** [comment]
你给我个联系方式，电报也行

**Comment by XIU2** [comment]
你去装个 ToDesk。
或者下个向日葵也行，都不需要注册登录。

然后把远程协助用的设备代码告诉我就行了。

**Comment by wwwliutao125** [comment]
我有

**Comment by wwwliutao125** [comment]
向日葵

**Comment by XIU2** [comment]
把本机识别码告诉我。

**Comment by wwwliutao125** [comment]
这里面不能私信么，加个好友也行

**Comment by XIU2** [comment]
如果后续没有出现其他问题的话，那我就先关闭该 Issues 了。
有问题可以继续在这 Issues 下告诉我。
